python-sdk/src/astro/__init__.py

```diff
@@ -13,14 +13,20 @@
 # imported beforehand, it will also load astro.sql. In astro.sql we import lots of operators which depend on
 # astro.database, and this is what leads to the circular dependency.
 
-import astro.sql  # noqa: F401
-from airflow.configuration import conf
-
-
 # This is needed to allow Airflow to pick up specific metadata fields it needs
 # for certain features. We recognize it's a bit unclean to define these in
 # multiple places, but at this point it's the only workaround if you'd like
 # your custom conn type to show up in the Airflow UI.
+
+# To prevent this from being a breaking change, we have to add our backend to taskinstance explicitly
+# The alternative would be to force users to set our custom xcom backend in their airflow configs, which would
+# be a breaking change which would require a 2.0 release
+from airflow.models import taskinstance
+from astro.custom_backend.astro_custom_backend import AstroCustomXcomBackend
+
+taskinstance.XCom = AstroCustomXcomBackend
```

Comment:
I'm curious if Airflow has any alternative, cleaner way for a library to redefine the default `XCom` backend. I would love to have feedback from other Airflow committers @kaxil @ashb @feluelle